docs/content/practices/network.md

```diff
@@ -1,5 +1,148 @@
+  Also, we've got an opportunity to easily mock authorisation in this mode, as we've already had all the data we need to put into preferences/stores_
+
+<Picture2 TBD>
+
+
+**Instruments:**
+
+* **[Wiremock](https://github.com/wiremock/wiremock)**   
+     ➕ Works on the system level (`MITM` internally)
+     <br>➕ `Record/Replay` from the box
+     <br>➕ Keeping request time in `playback` mode 
+     <br>➕ Request indexing _(the same request can be recorded with different answers, if it executes multiple times)_ 
+     <br>➕ Smart request substitution _(Will substitute similar request if particular one hasn't been found)_
+     <br>➕ Manually mocking support
+     <br><br>➖ Not Android friendly
+     <br>➖ Doesn't support multiple hosts from the box
+     <br>➖ Doesn't convenient output (set of json files)
+     <br>➖ Requires a custom implementation of storing files
+    <br><br>You may find an example [how to implement it here](https://handstandsam.com/2016/01/30/running-wiremock-on-android/)
+     
+  
+  
+* **[OkReplay](https://github.com/airbnb/okreplay)**
+  <br>➕ Android Friendly, easy adoption
+  <br>➕ OkHttp interceptor internally
+  <br>➕ Multiple hosts from the box (because of interceptor)
+  <br>➕ Readable and editable output _([.yaml file](https://github.com/airbnb/okreplay/blob/master/okreplay-tests/src/test/resources/okreplay/tapes/smoke_spec.yaml) with all recorded requests and responses in a one place)_
+  <br>➕ Opportunity to pull stubs from the device and store them to the device
+  <br><br>➖ Doesn't maintain
+  <br>➖ Doesn't have request indexing
+  <br>➖ Doesn't keep http request time
+  <br>➖ Doesn't have manually mocking
+  <br>➖ Doesn't have smart indexing
+  
+  
+* **Solution from Revolut (Work in progress)**
+  <br>_Revolut prepares an open source library, which allows getting rid of all problems mentioned above and make adoption a way easier_
+
+**Problems**<br>
+
+* Account preparation
+  <br>_You need to prepare a user which matches the special condition you need to test. It can be still mocked manually or via specially created service, which does this job for you_
+  <br>
+* Core changes
+  <br>_As soon as your root code base changed in terms of http requests, you probably would need to re-record some tests. That's why it's better to skip authorisation mode in the `playback` mode, to reduce such cases_
+  
+
+### Conclusion
+
+There is no perfect solution exists, it all depends on a number of factors, for example, the size of the team and how often the code base changes.
+It also depends on what goals you set in UI testing.
+
+??? question "Should I avoid real network testing in general?"
+    Depends on UI testing goals, project type and testing priorities.
+    <br> <br> If it's important for you to have UI testing as a part of `Pull Request` or `Release` pipelines, then it's very important to put the stability of these tests first
```

Comment:
I would add some enumeration here since these are very good points and need some highlighting